src/routes/transactions/mappers/common/native-staking.mapper.ts

```diff
@@ -168,6 +179,66 @@ export class NativeStakingMapper {
     });
   }
 
+  /**
+   * Maps the {@link NativeStakingWithdrawTransactionInfo} for the given
+   * native staking deployment's `withdraw` call
+   *
+   * @param args.chainId - the chain ID of the native staking deployment
+   * @param args.to - the address of the native staking deployment
+   * @param args.value - the value of the withdraw transaction
+   * @param args.transaction - the transaction object for the withdraw
+   * @returns {@link NativeStakingWithdrawTransactionInfo} for the given native staking deployment
+   */
+  // TODO: refactor common logic between mapDepositInfo and mapValidatorsExitInfo and mapWithdrawInfo
+  public async mapWithdrawInfo(args: {
+    chainId: string;
+    to: `0x${string}`;
+    value: string | null;
+    transaction: MultisigTransaction | ModuleTransaction | null;
+  }): Promise<NativeStakingWithdrawTransactionInfo> {
+    const [chain, deployment] = await Promise.all([
+      this.chainsRepository.getChain(args.chainId),
+      this.stakingRepository.getDeployment({
+        chainId: args.chainId,
+        address: args.to,
+      }),
+    ]);
+
+    if (
+      deployment.product_type !== 'dedicated' ||
+      deployment.chain === 'unknown' ||
+      deployment.status === 'unknown'
+    ) {
+      throw new NotFoundException('Native staking deployment not found');
+    }
+
+    let value = Number(args.value ?? 0);
+
+    if (args.transaction) {
+      const validatorsPublicKeys = args.transaction?.dataDecoded
+        ? this.getPublicKeysFromDataDecoded(args.transaction.dataDecoded)
+        : [];
+
+      const stakes = await this.stakingRepository.getStakes({
```

Comment:
We don't need to make this call if `validatorsPublicKeys` has no length.

Reply:
Fixed in [1c72417](https://github.com/safe-global/safe-client-gateway/pull/1882/commits/1c72417d15f4ca88ccf9639d88d82ed0721d2ec5)

Thanks!

Reply:
I [don't see the change](https://github.com/safe-global/safe-client-gateway/blob/f8bd2ceab83ab2bfc068e06ac7672e30df21dec0/src/routes/transactions/mappers/common/native-staking.mapper.ts#L218-L222).